src/main/java/org/wildfly/security/pem/Pem.java

```diff
@@ -185,9 +250,25 @@ public static void generatePemX509Certificate(ByteStringBuilder target, X509Cert
         Assert.checkNotNullParam("target", target);
         Assert.checkNotNullParam("certificate", certificate);
         try {
-            generatePemContent(target, "CERTIFICATE", ByteIterator.ofBytes(certificate.getEncoded()));
+            generatePemContent(target, CERTIFICATE_FORMAT, ByteIterator.ofBytes(certificate.getEncoded()));
         } catch (CertificateEncodingException e) {
             throw log.certificateParseError(e);
         }
     }
+
+    /**
+     * Generate PEM content containing a {@link PublicKey}.
+     *
+     * @param target the target byte string builder (must not be {@code null})
+     * @param publicKey the {@link PublicKey} (must not be {@code null})
+     */
+    public static void generatePemPublicKey(ByteStringBuilder target, PublicKey publicKey) {
+        Assert.checkNotNullParam("target", target);
+        Assert.checkNotNullParam("publicKey", publicKey);
+        String format = publicKey.getFormat();
+        if (!"X.509".equals(format)) {
```

Comment:
You could use a KeyFactory to get an X509EncodedKeySpec, and not have to rely on a specific format for the public key.